Add 2.8.5 release notes (#8761)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## Release 2.8.5

### Enhancements

> - Add basic event instrumentation to blocks — https://github.com/PrefectHQ/prefect/pull/8686

Reviewer: Experimental

- Add an endpoint to retrieve data from the collection registry — https://github.com/PrefectHQ/prefect/pull/8685

### Fixes

> - Add carriage return to `prompt_select_from_list` — https://github.com/PrefectHQ/prefect/pull/8705

Reviewer: ```suggestion
- Fix `prefect cloud login` detection of "ENTER" on some machines — https://github.com/PrefectHQ/prefect/pull/8705
```

> - Fix Kubernetes job watch timeout request error by rounding — https://github.com/PrefectHQ/prefect/pull/8733

Reviewer: ```suggestion
- Fix Kubernetes job watch timeout request error by rounding floats — https://github.com/PrefectHQ/prefect/pull/8733
```

> - Ignore fsspec `2023.3.0` during requirements installation — https://github.com/PrefectHQ/prefect/pull/8757

Reviewer: ```suggestion
- Fix flow load errors by ignoring fsspec `2023.3.0` during requirements installation — https://github.com/PrefectHQ/prefect/pull/8757
```

> - Use `PREFECT_API_URL` when initializing the events client — https://github.com/PrefectHQ/prefect/pull/8704

Reviewer: Experimental

- Fix Deployment and Concurrency Limit pages tabs — https://github.com/PrefectHQ/prefect/pull/8716

> - Add tests for base exceptions and calls — https://github.com/PrefectHQ/prefect/pull/8734

Reviewer: Maintenance
```suggestion
```

> - Remove deployment flow run foreign key — https://github.com/PrefectHQ/prefect/pull/8684

Reviewer: Arguably an enhancement not a fix; what's the user-facing effect? Improve deletion of deployments?

Reviewer: @rpeden are we going to include more details about the ux here?

Author: yes, added in latest commit

### Experimental
- Refactor supervisor API to allow configuration — https://github.com/PrefectHQ/prefect/pull/8695
- Consolidate `WorkItem` and `Call` classes — https://github.com/PrefectHQ/prefect/pull/8697
- Refactor supervisors to interact directly with "Worker" threads — https://github.com/PrefectHQ/prefect/pull/8714
- Add chaining to cancel contexts — https://github.com/PrefectHQ/prefect/pull/8719
- Add portal abstract base for worker threads and supervisors — https://github.com/PrefectHQ/prefect/pull/8717
- Fix bugs in supervisors implementation — https://github.com/PrefectHQ/prefect/pull/8718
- Refactor concurrency module and add documentation — https://github.com/PrefectHQ/prefect/pull/8724
- Update block event resource IDs to use block-document id instead of name. — https://github.com/PrefectHQ/prefect/pull/8730

> - UI updates for workers beta — https://github.com/PrefectHQ/prefect/pull/8732

Reviewer: Beta cancelled
```suggestion
```
